### DISCUSSION TITLE
Different skeletons in one model?

### DISCUSSION THREAD

**Discussion by janiceylau** [talmolab/sleap#1257] [Help!] [answered]
Hello! I am going to label some cows from 3 different angles: from the top, from its left and from its right. So I need 3 different skeletons to do this. It is more about the areas than individual points so I need 3 skeletons. Is it possible to have one labelled file with 3 skeletons? 
Or another possibility is to label in this way:  nodes 1 to 4 for the first skeleton, 5 to 8 for the second, 9 to 12 for the third (though in this case there is actually only 1 skeleton). I wonder if it is possible to train a model in this way. Or is it supposed that most of the time we could label all the nodes on each animal?

**Comment by roomrys** [comment, marked as answer]
Hi @janiceylau,

SLEAP currently supports just a single skeleton per slp file. You could create a big skeleton to label nodes in the divided manner you mention above, but, given the info above, my recommendation would be to create one project per each unique skeleton. Having the skeleton split-up will modularize your goal and you can retrain a model for that particular view without needing to guess which views need more labels.

For the different views, do you have a multi-camera set-up (where each view is from a new angle/camera/video)? If the views are indeed separated (each video contains only a single view), then I agree with my recommendation above. You could separate the project, adding only the videos containing a particular view to respective projects. However, if the different views can be seen from the same video, then it might make more sense to have one large skeleton and mark some nodes as non-visible as the cow gradually moves from view to view.

Let us know if this helps!

Thanks,
Liezl

**Comment by janiceylau** [comment]
Hi Liezl, 

Thanks so much for your reply! Right each video contains only a single view. Now I know better about what to do!

A follow-up question: I will now use three separate projects and hence three separate label files (.slp file) for different angles. Is it possible to combine the three label files into one later in case I want to try out if that works?

Best,
Janice

**Reply by roomrys** [reply]
Hi @janiceylau,

You bring up a good point that I totally forgot to mention! Yes, so while we have a feature for merging slp files (including the skeletons), SLEAP does not have a feature for splitting slp files - another incentive for starting out with the separate projects. 

One important piece of info (that is easy to do/change) is that your skeletons should have unique node names if you would like the nodes to be unique upon merging (i.e. `skeleton_1=[head, torso, tail]` merged with `skeleton_2=[head, ear_l, ear_r]` will yield `skeleton_3=[head, torso, tail, ear_l, ear_r]` where the `head` is considered the same node/body-part in `skeleton_3`).

Thanks,
Liezl

**Comment by janiceylau** [comment]
Great! Thank you so much!
